test: use kata 3.6.0 for testing

littlejawa opened the pull request:
#### What type of PR is this?

/kind ci

#### What this PR does / why we need it:

Update the version of kata to test with - using the latest 3.6.0 version. 
Also make kata test specific changes (affecting only the fedora-kata job).

Details:
- move to kata 3.6.0
- enable hooks tests, as the fix for it was released with 3.5.0
- make sure the storage.overlay.skip_mount_home flag is set to true (problem introduced with kata 3.3.0)
- enable debug logs for kata
- cleanup lingering kata processes on teardown
- add an integration test specific to kata, testing we're actually using the kata runtime. Useful to validate config/script changes in the CI

#### Which issue(s) this PR fixes:

None

#### Special notes for your reviewer:

#### Does this PR introduce a user-facing change?

```release-note
None
```

kwilczynski commented:
/retest-required

kwilczynski commented:
/approve

kwilczynski commented:
/ok-to-test

kwilczynski commented:
/retest-required

littlejawa commented:
/hold

littlejawa commented:
/test ci-fedora-kata

kwilczynski commented:
/retest

kwilczynski commented:
/retest

kwilczynski commented:
/retest

littlejawa commented:
/unhold

kwilczynski commented on contrib/test/ci/build/kata.yml:
```diff
@@ -61,6 +61,14 @@
+
+- name: Set debugging logs
+  block:
+    - name: Edit Kata Containers config to enable debug logl level
```
A typo? :smile: 

```suggestion
    - name: Edit Kata Containers config to enable debug log level
```

littlejawa replied:
Fixed. Thanks ! :-)

kwilczynski commented:
/approve
/lgtm

kwilczynski commented:
/retest

kwilczynski commented:
/approve
/lgtm

kwilczynski commented:
/retest

littlejawa commented:
Update:
- move to kata 3.6.0
- enable hooks tests, as the fix for it was released with 3.5.0
- add some kata test specific enhancements (enable debug, cleanup lingering processes)
- add an integration test specific to kata, testing we're actually using the kata runtime. Useful to validate config/script changes in the CI

littlejawa commented:
/retest

littlejawa commented:
/retest

littlejawa commented:
/retest

littlejawa commented:
/retest

littlejawa commented:
/retest

littlejawa commented:
@kwilczynski - sorry for pinging you again on this one, but I've updated it and it needs approval again.

Note that I have modified some settings for the kata test, but they have no impact on other CI jobs.

kwilczynski commented:
/approve 
/lgtm

kwilczynski commented:
/approve
/lgtm

kwilczynski commented:
@cri-o/cri-o-maintainers, please help a look and help us with some final approvals, thank you!

littlejawa commented:
/retest

littlejawa commented:
/test ci-cgroupv2-integration

littlejawa commented:
/test e2e-gcp-ovn

littlejawa commented:
/retest

littlejawa commented:
/test ci-fedora-integration

littlejawa commented:
/test ci-fedora-integration

littlejawa commented:
@cri-o/cri-o-maintainers - I think this one is ready to merge. The remaining failure on fedora-integration is unrelated as far as I can tell.

saschagrunert commented on test/kata.bats:
```diff
@@ -0,0 +1,47 @@
+	cleanup_test
+}
+
+#@test "kata: crio config should have the default runtime set to kata" {
+#	start_crio
+#
+#	# make sure we're using kata as the default runtime
+#	output="$($CRIO_BINARY_PATH status config 2>/dev/null | grep default_runtime)"
+#	echo $output
+#	[[ "$output" == *'default_runtime = "kata"'* ]]
+#}
```
Can we remove that?

saschagrunert commented:
/retest

littlejawa commented:
/retest

littlejawa commented:
@saschagrunert - remaining error in cgroupv2-integration is unrelated (see #8389)
And I know you're working on fixing the RPM builds :-)
Do you think we can merge it?

saschagrunert commented:
/override ci/prow/ci-cgroupv2-integration

saschagrunert commented:
Merging, the kata tests were successful.
